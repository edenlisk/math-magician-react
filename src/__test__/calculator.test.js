import TestRenderer from 'react-test-renderer';
import Calculator from '../components/calculator';

describe('Test Calculator', () => {
  test('It should render Calculator component correctly', () => {
    const tree = TestRenderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});