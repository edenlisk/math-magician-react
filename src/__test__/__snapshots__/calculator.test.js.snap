// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calculator It should render Calculator component correctly 1`] = `
<div
  className="container d-flex justify-content-center py-5"
>
  <div
    className="flex-fill"
  >
    <h5>
      Let's do the math!
    </h5>
  </div>
  <div
    className="flex-fill border border-3 border-dark justify-content-center py-3"
  >
    <div
      className="calculator-container text-center mx-auto"
    >
      <p
        className="result-field border-0 mx-auto text-end text-white fs-3 fw-bold my-0"
      />
      <div
        className="row btn-container text-center mx-auto"
      >
        <button
          className="btn btn-outline-secondary rounded-0 col"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="btn btn-outline-secondary rounded-0 col"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="btn btn-outline-secondary rounded-0 col"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="rounded-0 border-secondary operator col"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="row btn-container text-center mx-auto"
      >
        <button
          className="btn btn-outline-secondary rounded-0 col"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="btn btn-outline-secondary rounded-0 col"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="btn btn-outline-secondary rounded-0 col"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="rounded-0 border-secondary col operator"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="row btn-container text-center mx-auto"
      >
        <button
          className="btn btn-outline-secondary rounded-0 col"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="btn btn-outline-secondary rounded-0 col"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="btn btn-outline-secondary rounded-0 col"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="rounded-0 border-secondary col operator"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="row btn-container text-center mx-auto"
      >
        <button
          className="btn btn-outline-secondary rounded-0 col"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="btn btn-outline-secondary rounded-0 col"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="btn btn-outline-secondary rounded-0 col"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="rounded-0 border-secondary col operator"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="row btn-container text-center mx-auto"
      >
        <button
          className="btn btn-outline-secondary rounded-0 col-6"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="btn btn-outline-secondary rounded-0 col"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="rounded-0 border-secondary col operator"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
